src/components/input-text/input-text.tsx

```diff
@@ -0,0 +1,554 @@
+    this.setValue({
+      committing: true,
+      nativeEvent,
+      origin: "user",
+      value: ""
+    });
+  };
+
+  private emitChangeIfUserModified = (): void => {
+    if (this.previousValueOrigin === "user" && this.value !== this.previousEmittedValue) {
+      this.calciteInputTextChange.emit();
+    }
+    this.previousEmittedValue = this.value;
+  };
+
+  private inputBlurHandler = () => {
+    this.calciteInternalInputTextBlur.emit({
+      element: this.childEl,
+      value: this.value
+    });
+
+    this.emitChangeIfUserModified();
+  };
+
+  private inputFocusHandler = (event: FocusEvent): void => {
+    const slottedActionEl = getSlotted(this.el, "action");
+    if (event.target !== slottedActionEl) {
+      this.setFocus();
+    }
+    this.calciteInternalInputTextFocus.emit({
+      element: this.childEl,
+      value: this.value
+    });
+  };
+
+  private inputInputHandler = (nativeEvent: InputEvent): void => {
+    if (this.disabled || this.readOnly) {
+      return;
+    }
+    this.setValue({
+      nativeEvent,
+      origin: "user",
+      value: (nativeEvent.target as HTMLInputElement).value
+    });
+  };
+
+  private inputKeyDownHandler = (event: KeyboardEvent): void => {
+    if (this.disabled || this.readOnly) {
+      return;
+    }
+    if (event.key === "Enter") {
+      this.emitChangeIfUserModified();
+    }
+  };
+
+  onFormReset(): void {
+    this.setValue({
+      origin: "reset",
+      value: this.defaultValue
+    });
+  }
+
+  syncHiddenFormInput(input: HTMLInputElement): void {
+    if (this.minLength != null) {
+      input.minLength = this.minLength;
+    }
+
+    if (this.maxLength != null) {
+      input.maxLength = this.maxLength;
+    }
+  }
+
+  private setChildElRef = (el) => {
+    this.childEl = el;
+  };
+
+  private setDisabledAction(): void {
+    const slottedActionEl = getSlotted(this.el, "action");
+
+    if (!slottedActionEl) {
+      return;
+    }
+
+    this.disabled
+      ? slottedActionEl.setAttribute("disabled", "")
+      : slottedActionEl.removeAttribute("disabled");
+  }
+
+  private setInputValue = (newInputValue: string): void => {
+    if (!this.childEl) {
+      return;
+    }
+    this.childEl.value = newInputValue;
+  };
+
+  private setPreviousEmittedValue = (newPreviousEmittedValue: string): void => {
+    this.previousEmittedValue = newPreviousEmittedValue;
+  };
+
+  private setPreviousValue = (newPreviousValue: string): void => {
+    this.previousValue = newPreviousValue;
+  };
+
+  private setValue = ({
+    committing = false,
+    nativeEvent,
+    origin,
+    previousValue,
+    value
+  }: {
+    committing?: boolean;
+    nativeEvent?: MouseEvent | KeyboardEvent | InputEvent;
+    origin: SetValueOrigin;
+    previousValue?: string;
+    value: string;
+  }): void => {
+    this.setPreviousValue(previousValue || this.value);
+    this.previousValueOrigin = origin;
+    this.userChangedValue = origin === "user";
```

Comment:
☝️ any reason why the `&& this.value !== newValue` didn't carry over on this line?  I recall adding that because there were sometimes cases when the new value coming into setValue was the same as the previous value so the user couldn't possibly have changed it in that case since it is the same as the previous value.

Reply:
Because we don't need `newValue` anymore. For type="text" it looks like `value === newValue`

Reply:
Right, I forgot that `newValue` was calculated as part of the number input.  I think it would be good to check that the value coming into `setValue` is actually different than the currently set `this.value`, particularly for this state property, so it could read `this.userChangedValue = origin === "user" && value !== this.value`.